Exam/CC1/src/exercice4/ReservationManager.java

```diff
@@ -0,0 +1,52 @@
+package exercice4;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Hashtable;
+import java.util.Set;
+
+public class ReservationManager {
+	private final Hashtable<Integer, Reservation> reservations = new Hashtable<Integer, Reservation>();
+	
+	public Reservation getReservation(Integer num) {
+		return reservations.get(num);
+	}
+	public Reservation getReservationDate(String date) throws ParseException {
+		SimpleDateFormat f = new SimpleDateFormat("dd/MM/yyyy"); 
+		Date dateFind = f.parse(date);
+		Set<Integer> keys = reservations.keySet();
+        for(Integer key: keys){
```

Comment:
Why iterate over keys, when you can iterate over values?
~~~java
for(Reservation r: reservations.values())
~~~